### PR TITLE
Add a "Compile Application and OpenLayers 3 Together" tutorial

### PR DESCRIPTION
**Pull request by elemoine**
This tutorial requires https://github.com/openlayers/closure-util/pull/53 and https://github.com/openlayers/closure-util/pull/52.

This is the repo corresponding to the project built in the tutorial: https://github.com/elemoine/openlayers-closure-application/.

### CODE REVIEW

**Comment by bartvde** [comment]
@elemoine is this ready for review or should I wait with reviewing?

**Comment by elemoine** [comment]
The text can be reviewed. But the tutorial itself requires merging the closure-util PRs and publishing a new closure-util version. Thanks!

File: doc/tutorials/compile-application.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+You are done!
+
+## Run Application in Debug Mode
+
+As a bonus, we're going to show how to use `closure-util` to run the
+application in « debug » mode, where the original application and OpenLayers
+scripts are loaded one by one in the page.
+
+Start the `clorsure-util` development server:
```
Comment (rdewit) [inline doc/tutorials/compile-application.md]: There is a minor typo in `clorsure-util`. It should be `closure-util`

File: doc/tutorials/compile-application.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+---
+
+# Compiling Application with Closure Compiler
+
+The OpenLayers 3 code uses the Closure Library, and it is compiled with the
+Closure Compiler. Using OpenLayers 3 in an application does not require using
+Closure. But using Closure in an OpenLayers 3 is possible. And this is what
+this tutorial is about.
+
+This tutorial will teach you how to set up a OpenLayers 3 application based on
```
Comment (gberaudo) [inline doc/tutorials/compile-application.md]: an

File: doc/tutorials/compile-application.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
+---
+title: Compiling Application with Closure Compiler
+layout: doc.hbs
+---
+
+# Compiling Application with Closure Compiler
+
+The OpenLayers 3 code uses the Closure Library, and it is compiled with the
+Closure Compiler. Using OpenLayers 3 in an application does not require using
+Closure. But using Closure in an OpenLayers 3 is possible. And this is what
```
Comment (gberaudo) [inline doc/tutorials/compile-application.md]: Do you mean an Openlayers 3 `application`?

File: doc/tutorials/compile-application.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+You don't need to read the `closure-util` documentation to follow this
+tutorial, but it may be useful if you feel the need for more information.
+
+## Advantages of using Closure
+
+This tutorial will show you how to use the Closure Compiler to compile an
+application and OpenLayers 3 together. Compiling the application code together
+with the OpenLayers 3 code has a number of advantages.
+
+Fist of all, it allows you to only « pay » for the OpenLayers 3 code your
```
Comment (marcjansen) [inline doc/tutorials/compile-application.md]: Fi **r** st, not Fist

File: doc/tutorials/compile-application.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+application uses, as the compiler will exclude the OpenLayers 3 that the
```
Comment (marcjansen) [inline doc/tutorials/compile-application.md]: ...exclude the OpenLayers 3 **code** that...

**Comment by marcjansen** [comment]
A very nice tutorial! I added minor comments to certain lines.

File: doc/tutorials/compile-application.md (hunk 4 above)
Comment (gberaudo) [inline doc/tutorials/compile-application.md]: Typo: First

File: doc/tutorials/compile-application.md (hunk 5 above)
Comment (gberaudo) [inline doc/tutorials/compile-application.md]: Missing word: code

File: doc/tutorials/compile-application.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+application doesn't use. And there's no need to write and maintain a list of
```
Comment (gberaudo) [inline doc/tutorials/compile-application.md]: there is

File: doc/tutorials/compile-application.md [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+
+Now change to that directory:
+
+    $ cd openlayers-closure-application
+
+Our application will be a `node` application, and the `openlayers` and
+`closure-util` node packages will be downloaded from the node package registry
+using the `npm` command line tool.
+
+So we're going to create a `package.json` file for the application, which the
```
Comment (gberaudo) [inline doc/tutorials/compile-application.md]: with

File: doc/tutorials/compile-application.md [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+```
+
+Create a `config.json` file with the above content at the root of the
+application directory.
+
+We can now use `closure-util` to compile the code:
+
+    $ ./node_modules/openlayers/node_modules/.bin/closure-util build config.json app.js
+
+The resulting `app.js` file, which you can edit in your editor if you're
```
Comment (gberaudo) [inline doc/tutorials/compile-application.md]: view

File: doc/tutorials/compile-application.md [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+  </head>
+  <body>
+    <div id="map"></div>
+    <script src="app.js" type="text/javascript"></script>
+  </body>
+</html>
+```
+
+Note that the page includes a `script` tag referencing the `app.js` file,
+that resulting from the compilation.
```
Comment (gberaudo) [inline doc/tutorials/compile-application.md]: Use `the one` instead of `that`?

**Comment by elemoine** [comment]
Thank you all for your comments. I think I addressed them all.

As I said in the issue description I created an `openlayers-closure-application` Git [repository](https://github.com/elemoine/openlayers-closure-application/) including the application built in this tutorial. The tutorial should include a link to the that repository. The repo is in my GitHub space for the moment. Do you guys think it should be moved to http://github.com/openlayers? I could also rename the repo to `ol3-closure-sample-app` or something.

**Comment by bartvde** [comment]
hey @elemoine I think it would make sense to have that repo in the openlayers organisation if that's okay with you

**Comment by elemoine** [comment]
Ok, thanks, I'll add it and I'll add a reference to it in the tutorial. I'm still waiting for https://github.com/openlayers/closure-util/pull/53 to be reviewed and merged before merging this one.

**Pull request merged**